include/aspect/material_model/multicomponent.h

```diff
@@ -152,6 +124,12 @@ namespace aspect
          */
 
         /**
+         * Make vector double of parameters
+         */
+        virtual std::vector<double> get_vector_double (const std::string &parameter, unsigned int &n_fields, ParameterHandler &prm);
```

Comment:
Out of curiosity, why do you make this function (and some others below) `virtual`? This seems unnecessary.